go/genkit/flow.go

```diff
@@ -117,26 +117,32 @@ type flowOptions struct {
 
 type noStream = func(context.Context, struct{}) error
 
-// FlowAuth configures a auth context provider and a auth policy check for a flow.
+// FlowAuth configures an auth context provider and a auth policy check for a flow.
 type FlowAuth interface {
-	// ProvideAuthContext provides auth context from an auth header.
-	ProvideAuthContext(ctx context.Context, authHeader string) (map[string]any, error)
+	// ProvideAuthContext provides auth context from an auth header and sets it on the context.
+	ProvideAuthContext(ctx context.Context, authHeader string) (context.Context, error)
+
+	// SetAuthContext sets the auth context on the given context.
+	SetAuthContext(ctx context.Context, authContext map[string]any) context.Context
```

Comment:
This I don't really understand. Contexts are supposed to behave like functional values; you shouldn't modify them or anything inside them. If this method is supposed to do that, then can we drop it in favor of the above? If not, why not?

Reply:
There are two cases we're trying to cover:

1) The prod case: this is where we're given an auth token in the header and we need to make a real API call to decode it into an auth context. This is what `ProvideAuthContext` does and it returns a `context.Context` for use both by the plugin to check policy and also to propagate into flow code so that the details may be used for other calls.
2) The dev/direct run case: this is where we're not making a real API call to auth but we need to provide the auth context either from the HTTP request or directly in flow code (direct run). This auth context needs to similarly be set on `context.Context` so that it may propagate through to flow code for later use if necessary.

That's why we have `SetAuthContext` because only the plugin knows what the key it's expecting is for both policy check and for `GetAuthContext`.

Reply:
What if SetAuthContext was called NewContext (since IIUC it is just using something it has locally) and we leave ProvideAuthContext?

Reply:
In either case, more doc please!

Reply:
Done, please take a final look to make sure it makes sense.

Reply:
LGTM.